src/Service/NotificationService.php

```diff
@@ -74,37 +78,37 @@ public function insertNotification(array $notification): void
     {
         //TODO Handle the notification as a model object to avoid this?
         $fields = [];
-        if (!empty($notification['pspReference'])) {
+        if (array_key_exists('pspReference', $notification)) {
             $fields['pspreference'] = $notification['pspReference'];
         }
-        if (!empty($notification['originalReference'])) {
+        if (array_key_exists('originalReference', $notification)) {
             $fields['originalReference'] = $notification['originalReference'];
         }
-        if (!empty($notification['merchantReference'])) {
+        if (array_key_exists('merchantReference', $notification)) {
             $fields['merchantReference'] = $notification['merchantReference'];
         }
-        if (!empty($notification['eventCode'])) {
+        if (array_key_exists('eventCode', $notification)) {
             $fields['eventCode'] = $notification['eventCode'];
         }
-        if (!empty($notification['success'])) {
+        if (array_key_exists('success', $notification)) {
             $fields['success'] = "true" === $notification['success'];
         }
-        if (!empty($notification['paymentMethod'])) {
+        if (array_key_exists('paymentMethod', $notification)) {
             $fields['paymentMethod'] = $notification['paymentMethod'];
         }
-        if (!empty($notification['amount']['value'])) {
-            $fields['amountValue'] = (string)$notification['amount']['value'];
+        if (array_key_exists('value', $notification['amount'] ?? [])) {
```

Comment:
isn't it going to throw an exception if the `$notification['amount']` does not exist?
Maybe better to use isset() instead of array_key_exists()
(isset though will return false if the value is null)

Reply:
I used the null coalesce operator (`?? []`) so that it doesn't throw an exception, this way it will use the empty array if the `amount` key is missing and then array_key_exists will return false. Because this is a nested array, with isset() I would have to do something like `if (isset($notification['amount']) && isset($notification['amount']['value']))`

Reply:
@peterojo isset() handles multi level check as well and handles if the first key doesn't even exist so you could do only:
`isset($notification['amount']['value'])`

Reply:
You're right, I can change it all to `isset()`. I changed from `!empty()` because of `$notification['live']` which could have a value of `false` and `!empty()` would return false. But `isset()` also works correctly with this scenario